runtime/auth/aws-credentials/common/src/aws/smithy/kotlin/runtime/auth/awscredentials/CredentialsProvider.kt

```diff
@@ -4,12 +4,19 @@
  */
 package aws.smithy.kotlin.runtime.auth.awscredentials
 
+import aws.smithy.kotlin.runtime.io.Closeable
+
 /**
  * Represents a producer/source of AWS credentials
  */
-public interface CredentialsProvider {
+public interface CredentialsProvider : Closeable {
```

Comment:
i would argue it simplifies the implementation (going back to the point that there does exist a sensible `close()` default behavior which is to no-op) but i don't have a strong opinion

Reply:
With the introduction of `SdkManaged` I'm a much stronger advocate of having this implement `Closeable` by default:
* same reason as before w.r.t. no-op `close`
* `DefaultCredentialsProvider` itself implements `Closeable`. It makes perfect sense characteristically (in my mind) that a credentials provider might have some lifecycle with resources that need to be explicitly reaped(just like `HttpClientEngine` - it might _not_ need an explicit `close()` on a platform like js where management of connections/resources is likely entirely internal to the node runtime, but we mark it as implementing close anyway rather than just say having `OkHttpEngine` specifically be closeable)
* simplifies the implementation when wrapping into `SdkManagedCloseable` - if the default `close` wasn't there, we'd have to check and wrap with a no-op implementation